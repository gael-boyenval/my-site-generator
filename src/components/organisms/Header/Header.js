import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Link from 'gatsby-link';

import Grid from '../../layout/Grid';
import { TinyHeading } from '../../atoms/Typography';
import { spaces, unit, family } from '../../../styles/tokens';
import { rem } from '../../../styles/utils';

const Menu = styled.ul`
  list-style-type: none;
  text-align: right;
`;

const Wrapper = styled.header`
  height: ${rem(unit * 8)};
  display: flex;
  padding: ${rem(spaces.l)};
`;

const Item = TinyHeading.withComponent('li').extend`
  display: inline-block;
  padding-left: ${rem(spaces.m)};
`;

const Logo = styled.h3`
  font-family: ${family.sans};
  font-weight: 900;
  display: inline-block;
`;
const Hidden = styled.span`
  visibility: hidden;
  font-size: 0;
  text-indent: 100vh;
`;

const Header = ({ data }) => (
  <Wrapper>
    <Grid center gutter>
      <Grid.Cell>
        <Link to="/">
          <Logo>
            G.<Hidden>{data.site.siteMetadata.title}</Hidden>
          </Logo>
        </Link>
      </Grid.Cell>
      <Grid.Cell width="fill">
        <nav>
          <Menu>
            <Item>
              <Link to="/page2/">Page 2</Link>
            </Item>
            <Item>
              <Link to="/counter/">Counter</Link>
            </Item>
            <Item>
              <Link to="/my-files/">files</Link>
            </Item>
          </Menu>
        </nav>
      </Grid.Cell>
    </Grid>
  </Wrapper>
);

Header.propTypes = {
  data: PropTypes.shape({}),
};

Header.defaultProps = {
  data: {},
};

export default Header;
