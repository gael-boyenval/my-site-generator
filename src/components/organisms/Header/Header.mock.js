export default {
  site: {
    siteMetadata: {
      title: 'lorem ipsum',
    },
  },
};
