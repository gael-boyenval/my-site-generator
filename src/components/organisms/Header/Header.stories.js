/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Header from './Header';
import mock from './Header.mock';

storiesOf('04 - Organisms', module).add('Header', withInfo()(() => <Header data={mock} />));
