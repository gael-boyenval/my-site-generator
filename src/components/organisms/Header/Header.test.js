/* eslint-disable no-undef */
import React from 'react';
import Header from './Header';

describe('Component Header', () => {
  const mockData = {
    site: {
      siteMetadata: {
        title: 'lorem ipsum',
      },
    },
  };

  it('should be render without crashing', () => {
    const component = shallow(<Header data={mockData} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
