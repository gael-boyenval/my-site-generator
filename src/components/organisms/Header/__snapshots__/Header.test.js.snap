// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Header should be render without crashing 1`] = `
<styled.header>
  <styled.div
    center={true}
    gutter={true}
  >
    <styled.div>
      <GatsbyLink
        to="/"
      >
        <styled.h3>
          G.
          <styled.span>
            lorem ipsum
          </styled.span>
        </styled.h3>
      </GatsbyLink>
    </styled.div>
    <styled.div
      width="fill"
    >
      <nav>
        <styled.ul>
          <styled.li>
            <GatsbyLink
              to="/page2/"
            >
              Page 2
            </GatsbyLink>
          </styled.li>
          <styled.li>
            <GatsbyLink
              to="/counter/"
            >
              Counter
            </GatsbyLink>
          </styled.li>
          <styled.li>
            <GatsbyLink
              to="/my-files/"
            >
              files
            </GatsbyLink>
          </styled.li>
        </styled.ul>
      </nav>
    </styled.div>
  </styled.div>
</styled.header>
`;
